### PR TITLE
Fixed bug: now selects first favorite item, on arrow key switch

### LINKED ISSUE #23: Keyboard keys < and > do not work to switch between history and favorites for first favourite
How to replicate?

Clear your favorites, add a text to favorites and use < and > to switch. It won't work. However, once you press the text in favorites then you can use < and > to switch.

### PR DESCRIPTION
**Pull request by R4H33M**
Fixes #23 
Replicated and fixed. Switching between the favorite list and clips list is more consistent now.

### CODE REVIEW

**Comment by emily0618** [comment]
Hmm, I can't reproduce the bug...

**Comment by humdingerb** [comment]
This doesn't fix the real issue.  It doesn't matter how many items are in the favourites list, if no item was selected before, none will be selected when the FavView gets focus.
My suggestion: Whenever the FavView gets focus, select the first item if none is currrently selected.

**Comment by R4H33M** [comment]
There. I've implemented it exactly as suggested. Works well.
What do you say?

**Comment by humdingerb** [comment]
But what happens if the user hits the tab key to change the focus instead of the right cursor key? :)

**Comment by R4H33M** [comment]
Totally forgot about that. Thanks! Now it should work in all cases!

**Review by humdingerb — CHANGES_REQUESTED** [review]

File: FavView.cpp [hunk 1]
```diff
@@ -115,6 +115,11 @@ FavView::MakeFocus(bool focused)
 	// only signal FavView is focused when gaining focus
 	if (focused)
 		my_app->fMainWindow->SetHistoryActiveFlag(!focused);
+	
+	if (focused){
```
Comment (humdingerb) [inline FavView.cpp]: Also tiny code style violation, space between round and curlies. :)

File: FavView.cpp [hunk 2]
```diff
@@ -115,6 +115,11 @@ FavView::MakeFocus(bool focused)
...
+		if (BListView::CurrentSelection() < 0)
+			BListView::Select(0);
```
[the remark this answers is not in the document]
Comment (R4H33M) [reply FavView.cpp]: Good question... Actually I have no idea, I'm still learning C++
Since it works both ways, let's go for the more concise method.

**Comment by R4H33M** [comment]
Ready for review!

File: FavView.cpp [hunk 3]
```diff
@@ -112,9 +112,15 @@ FavView::MakeFocus(bool focused)
 {
 	BListView::MakeFocus(focused);
 
-	// only signal FavView is focused when gaining focus
-	if (focused)
+
```
Comment (humdingerb) [inline FavView.cpp]: Remove the duplicate empty line.

File: FavView.cpp [hunk 4]
```diff
@@ -112,9 +112,15 @@ FavView::MakeFocus(bool focused)
 {
 	BListView::MakeFocus(focused);
 
-	// only signal FavView is focused when gaining focus
-	if (focused)
+	
+	if (focused) {
+		// only signal FavView is focused when gaining focus
 		my_app->fMainWindow->SetHistoryActiveFlag(!focused);
+
```
Comment (humdingerb) [inline FavView.cpp]: Remove the two tabs in the empty line.

**Review by humdingerb — CHANGES_REQUESTED** [review]

**Review by humdingerb — APPROVED** [review]

**Pull request merged**